core/domain/translation_domain.py

```diff
@@ -18,9 +18,258 @@
+            elif (
+                entityTranslation.translations[
+                translatable_item.content_id].needs_update
+            ):
+                contents_which_need_translation.append(translatable_item)
+
+        return contents_which_need_translation
+
+
+class EntityTranslation:
+    """The EntityTranslation represents an EntityTransaltionsModel for a given
+    version of an entity in a given language.
+
+    Args:
+        entity_id: str. The id of the corresponding entity.
+        entity_type: str. The type of the corresponding entity.
+        entity_version: str. The version of the corresponding entity.
+        language_code: str. The language code for a corresponding entity.
+        translations: dict(str, TranslatedContent). The translated content.
+    """
+
+    def __init__(
+        self,
+        entity_id: str,
+        entity_type: str,
+        entity_version: int,
+        language_code: str,
+        translations: Dict[str, TranslatedContent]
+    ):
+        self.entity_id = entity_id
+        self.entity_type = entity_type
+        self.entity_version = entity_version
+        self.language_code = language_code
+        self.translations = translations
+
+
+class TranslatableContent:
+    """TranslatablesContent represents a content of a translatable object which
+    can be translated into multiple languages.
+
+    Args:
+        content_id: str. The id of the corresponding content.
+        content: str|list(str). The content which can be translated.
+        content_type: str. The type of the corresponding content.
+    """
+
+    def __init__(
+        self,
+        content_id: str,
+        content: str|List[str],
+        content_type: str
+    ) -> None:
+        self.content_id = content_id
+        self.content = content
+        self.content_type = content_type
+
+    @classmethod
+    def from_dict(
+        cls,
+        translatable_content_dict: TranslatableContentDict
+    ) -> TranslatableContent:
+        """Returns a TranslatableContent object from its dict representation.
+
+        Args:
+            translatable_content_dict: dict. Dict representation of
+                TranslatableContent object.
+
+        Returns:
+            TranslatableContent. The TranslatableContent object.
+        """
+        return cls(
+            translatable_content_dict['content_id'],
+            translatable_content_dict['content'],
+            translatable_content_dict['content_type'])
+
+    def to_dict(self) -> TranslatableContentDict:
+        """Returns the dict representation of TranslatableContent object.
+
+        Returns:
+            dict. The dict representation of TranslatableContent object.
+        """
+        return {
+            'content_id': self.content_id,
+            'content': self.content,
+            'content_type': self.content_type
+        }
+
+
+class TranslatedContent:
+    """Represents the translated contents of TranslatableContent
```

Comment:
Be careful, here and elsewhere, whether to use "content" or "contents". Please check all your names and docstrings in this PR thoroughly for that and ensure that each is representing the correct thing.

I think a consistent usage would be to use "content" for a single translatable field, and "contents" for a collection of these.